### ISSUE TITLE
Deno bench --unstable fails stating to add unstable flag (v1.21.2, aarch64-apple-darwin)

### ISSUE THREAD

**Issue by code-ape** [denoland/deno#14608]
## TL;DR

Deno command `bench` should work with the `--unstable` flag on v1.21.2 but when run with it I get an error telling it to add it. This is presumably a bug so I'm filing a ticket here. Let me know if you need any more info!

## Details

**Environment:** 2020 Apple M1 MacBook Pro 13" with a M1 CPU in it and Deno v1.21.2 [aarch64-apple-darwin] installed via homebrew (binary sha256sum `d19a09a2dec4bf9f9f82cea4b9b5b71ad56436dc2d969fe7c4b18c1e6a6c1b5a`)

I went to try out Deno's benchmark feature by copying the example from the docs ([link](https://deno.land/manual@v1.21.2/tools/benchmarker)). I followed them exactly creating a `url_bench.ts` file with the same contents as shown and then running `deno bench --unstable url_bench.ts`. I get the following error telling me to add the `--unstable` flag which has already been added to the command:

```
$ deno bench --unstable url_bench.ts
Check file:///Users/foo/dev/bar/url_bench.ts
error: TS2339 [ERROR]: Property 'bench' does not exist on type 'typeof Deno'. 'Deno.bench' is an unstable API.
Did you forget to run with the '--unstable' flag?
Deno.bench('URL parsing', () => {
     ~~~~~
    at file:///Users/foo/dev/bar/url_bench.ts:1:6
```

## Test file contents

```ts
Deno.bench('URL parsing', () => {
  new URL('https://deno.land');
});
```

**Comment by caspervonb** [comment]
Can't reproduce on v1.21.3:

```
foo/benches $ deno bench --unstable foo_bench.ts 
cpu: Apple M1
runtime: deno 1.21.3 (aarch64-apple-darwin)

file:///Users/caspervonb/Repositories/foo/benches/foo_bench.ts
benchmark      time (avg)             (min … max)       p75       p99      p995
------------------------------------------------- -----------------------------
foo/benches $ 
```

Nor on v1.21.2:
```
foo/benches $ deno bench --unstable foo_bench.ts
Check file:///Users/caspervonb/Repositories/foo/benches/foo_bench.ts
cpu: Apple M1
runtime: deno 1.21.2 (aarch64-apple-darwin)

file:///Users/caspervonb/Repositories/foo/benches/foo_bench.ts
benchmark      time (avg)             (min … max)       p75       p99      p995
------------------------------------------------- -----------------------------
```

Not sure how brew installs things, maybe try the install script instead.

```sh
curl -fsSL https://deno.land/install.sh | sh
```

**Comment by code-ape** [comment]
Found the root of the bug! This was running in a directory with a `deno.jsonc` file which had the following `compilerOptions.lib` declared. Benchmarking will run if: (a) the `lib` declaration is removed or (b) if `"deno.unstable"` is manually added to it. Is this expected behavior or should I update this issue accordingly?

```json
{
  "compilerOptions": {
    "lib": [ "deno.window" ],
  },
}
```
